src/platform/realtek/zephyr/ota/matter_ota.c

```diff
@@ -0,0 +1,480 @@
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "matter_ota.h"
+
+#include "dfu_common.h"
+#include <stdlib.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/util.h>
+
+LOG_MODULE_REGISTER(matter_ota, CONFIG_CHIP_APP_LOG_LEVEL);
+
+/** Little Endian array to uint16. */
+#define LE_ARRAY_TO_UINT16(u16, a)                                                                                                 \
```

Comment:
chip has EndianBuffer classes to read/write stuff. These macros are harder to follow. 

Any chance we can make this cpp and use classes?